metricflow/sql/optimizer/tag_required_column_aliases.py

```diff
@@ -55,6 +55,7 @@ def __init__(self, tagged_column_alias_set: TaggedColumnAliasSet) -> None:
             traverses the SQL-query representation DAG.
         """
         self._column_alias_tagger = tagged_column_alias_set
+        self._cte_alias_to_cte_node: Dict[str, SqlCteNode] = {}
```

Comment:
Nit: can we rename this to something more clear like `_required_aliases_for_cte_nodes`?